src/api/entities/Asset/index.ts

```diff
@@ -491,16 +486,28 @@ export class Asset extends Entity<UniqueIdentifiers, string> {
 
     const rawTicker = stringToTicker(ticker, context);
 
-    if (callback) {
-      return statistics.investorCountPerAsset(rawTicker, count => {
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises -- callback errors should be handled by the caller
-        callback(u64ToBigNumber(count));
-      });
+    const iuDisabled = await disableInvestorUniqueness(rawTicker);
+
+    const assembleResult = (balanceEntries: [StorageKey, Balance][]): BigNumber => {
+      const assetBalances = balanceEntries.filter(([, balance]) =>
+        balanceToBigNumber(balance).gt(new BigNumber(0))
+      );
+
+      return new BigNumber(assetBalances.length);
+    };
+
+    if (boolToBoolean(iuDisabled)) {
+      const balanceEntries = await balanceOf.entries(rawTicker);
+      return assembleResult(balanceEntries);
     }
 
-    const result = await statistics.investorCountPerAsset(stringToTicker(ticker, context));
+    const scopeIdEntries = await scopeIdOf.entries(rawTicker);
```

Comment:
There can be multiple entries for different identities with the same scope ID. You need to only consider unique ones